x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SearchableSnapshotAction.java

```diff
@@ -181,6 +287,39 @@ boolean isForceMergeIndex() {
         return steps;
     }
 
+    /**
+     * Resolves the prefix to be used for the mounted index depending on the provided key
+     */
+    String getRestoredIndexPrefix(StepKey currentKey) {
+        if (storageType == null) {
+            if (currentKey.getPhase().equals(TimeseriesLifecycleType.FROZEN_PHASE)) {
+                return PARTIAL_RESTORED_INDEX_PREFIX;
+            } else {
+                return FULL_RESTORED_INDEX_PREFIX;
+            }
+        }
+        switch (storageType) {
+            case FULL_COPY:
+                return FULL_RESTORED_INDEX_PREFIX;
+            case SHARED_CACHE:
+                return PARTIAL_RESTORED_INDEX_PREFIX;
+            default:
+                throw new IllegalArgumentException("unexpected storage type: " + storageType);
+        }
+    }
+
+    // Resolves the storage type from a Nullable to non-Nullable type
+    MountSearchableSnapshotRequest.Storage getConcreteStorageType(StepKey currentKey) {
```

Comment:
this could be static

Reply:
These make use of the `storageType` instance variable, while we could passed it in, I think that would make the code unnecessarily verbose where these methods are called. I think I'd prefer to keep it as non-static if that's okay with you.